### PR TITLE
Emergency Shutter Takes Time to Weld Shut

### PR DESCRIPTION
**Pull request by SleepyGemmy**
no more closing shutters and instawelding them. sorry.

### CODE REVIEW

**Comment by SleepyGemmy** [comment]
!review

**Comment by SleepyGemmy** [comment]
code broke with the changes so i reverted them.

**Comment by SleepyGemmy** [comment]
i hate git so much part 2

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -259,18 +259,27 @@
...
+		if(WT.isOn())
+			user.visible_message(
+				SPAN_WARNING("[user] begins welding [src] [blocked ? "open" : "shut"]."),
+				SPAN_NOTICE("You begin welding [src] [blocked ? "open" : "shut"]."),
+				SPAN_ITALIC("You hear a welding torch on metal.")
+			)
+			playsound(src, 'sound/items/welder.ogg', 50, 1)
+			if (!do_after(user, 2/C.toolspeed SECONDS, act_target = src, extra_checks = CALLBACK(src, .proc/is_open, src.density)))
+				return
+			if(!WT.remove_fuel(0,user))
```
Comment (Arrow768) [inline code/game/machinery/doors/firedoor.dm:274]: If it takes time it should also require some fuel to do that.

**Review by Arrow768 — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (SleepyGemmy) [reply code/game/machinery/doors/firedoor.dm]: fixed.

Comment (SleepyGemmy) [reply code/game/machinery/doors/firedoor.dm]: just playtested to be sure. the code as-is already removes 1 unit of welding fuel when welding the emergency shutter shut or open.

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**